lima: 1.2.2 -> 2.0.3

kachick opened the pull request:
Diff: https://github.com/lima-vm/lima/compare/v1.2.2...v2.0.2

Releases

- https://github.com/lima-vm/lima/releases/tag/v2.0.0
- https://github.com/lima-vm/lima/releases/tag/v2.0.1
- https://github.com/lima-vm/lima/releases/tag/v2.0.2

2.0.0 replaced a call to the external pkill command with an internal function.
This change conflicts with the current nixpkgs patch.

Supersedes #460087
Updates #458941

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

```console
> nix-build --attr pkgs.lima.passthru.tests
...
Running phase: installCheckPhase
Executing versionCheckPhase
Successfully managed to find version 2.0.1 in the output of the command /nix/store/430cb307aas1i4abd4zaa90f7h858abj-lima-2.0.1/bin/limactl --version
limactl version 2.0.1
Finished versionCheckPhase
WARN[0000] unable to determine host timezone, falling back to default value
INFO[0000] "templates/default.yaml": OK
building '/nix/store/qx8d39n3wr5cx6fcp45k43khkyg40m48-actual.drv'...
WARN[0000] unable to determine host timezone, falling back to default value
WARN[0000] unable to determine host timezone, falling back to default value
building '/nix/store/3r6a237qs50j4vfg1x481wcqrm3m5lny-equal-contents-limactl-also-detects-additional-guest-agents-if-specified.drv'...
Checking:
limactl also detects additional guest agents if specified
expected /nix/store/v59hvrd24rsxw8nd13fd408ps0m436bz-expected and actual /nix/store/ibb8wpwcaamrmahizgnywzp5s6ygshdn-actual match.
OK
/nix/store/lk41ppi1107k3ig6cgvxj1ppp4d48f1v-equal-contents-limactl-also-detects-additional-guest-agents-if-specified
/nix/store/7k7pvj3pmzqdkhc9lq05hn8asfr6l09l-equal-contents-limactl-only-detects-host-s-architecture-guest-agent-by-default
```

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [x] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [x] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [x] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

cc: @voanhduy1512
[Imported mentions](https://github.com/NixOS/nixpkgs/pull/460087#issuecomment-3508521405): @aaschmid, @tricktron @devusb 

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

kachick committed:
lima: 1.2.2 -> 2.0.1

Diff: https://github.com/lima-vm/lima/compare/v1.2.2...v2.0.1

Release: https://github.com/lima-vm/lima/releases/tag/v2.0.1

2.0.0 replaced a call to the external pkill command with an internal function.
This change conflicts with the current nixpkgs patch,
so we must remove the patch to prevent a build error.
Updates: a9b8ff8c8cae79f4f849840895a85b9f7af55545

2.0.0: Replace a pkill call:
  https://github.com/lima-vm/lima/commit/75079f679b196d06904de2a63c5add3948706e2f
2.0.1: Adds graceful shutdown support:
  https://github.com/lima-vm/lima/commit/e769391e72d5bc0438b16fcf5cf622a85cb4bc42

kachick commented on doc/release-notes/rl-2511.section.md:
```diff
@@ -212,6 +212,8 @@
 
 - `lima` package now only includes the guest agent for the host's architecture by default. If your guest VM's architecture differs from your Lima host's, you'll need to enable the `lima-additional-guestagents` package by setting `withAdditionalGuestAgents = true` when overriding lima with this input.
 
+- `lima` has been updated from `1.2.2` to `2.0.1`. This major update includes several breaking changes, such as `/tmp/lima` no longer being mounted by default.
```
It seems available now https://github.com/NixOS/nixpkgs/pull/461526

kachick commented:
Thanks for the link. To consider the behavior test, waiting for it to reach master is a good idea.
I will mark this PR as a draft again and watch the tracker: https://nixpkgs-tracker.ocfox.me/?pr=461615

kachick commented:
Reached to master branch

kachick committed:
lima: 2.0.1 -> 2.0.2

Diff: https://github.com/lima-vm/lima/compare/v2.0.1...v2.0.2

Release: https://github.com/lima-vm/lima/releases/tag/v2.0.2

msgilligan commented:
@kachick Now that the `release-25.11` branch is active, I believe this can target unstable and can be considered for back-porting later. As you suggested above:

> * We can't merge this PR until the merge restriction is unlocked, but we can backport lima 2.x into release-25.11.

The PR also needs a rebase.

kachick commented on doc/release-notes/rl-2605.section.md:
```diff
@@ -13,6 +13,8 @@
 
 - All Log4Shell vulnerability scanners were removed, as they were all unmaintained upstream and are no longer relevant given that the vulnerability has been fixed upstream for several years.
 
+- `lima` has been updated from `1.2.2` to `2.0.x`. This major update includes several breaking changes, such as `/tmp/lima` no longer being mounted by default.
```
https://github.com/NixOS/nixpkgs/pull/461178#issuecomment-3650486659

kachick commented:
Yes.

kachick committed:
lima: 2.0.2 -> 2.0.3

Diff: https://github.com/lima-vm/lima/compare/v2.0.2...v2.0.3

Release: https://github.com/lima-vm/lima/releases/tag/v2.0.3
